### ISSUE TITLE
Impossible customization by CSS

### ISSUE THREAD

**Issue by MatsMaker** [markusenglund/react-switch#63]
We need to add customization styles from file styles. And not use only inline styles in the component props.

**Comment by CesarGonzAndres** [comment]
True... CSS customization is impossible... I'm facing the same issue

**Comment by BorisL91** [comment]
Is there any update on this? Thanks!

**Comment by sabahath786** [comment]
I have same question, recently i started using this library, i would like to know how can we use separate css file for this component instead of inline css

**Comment by santisiri** [comment]
css styles is a must :(

**Comment by markusenglund** [comment]
Due to how the component is built, CSS can't be used for some properties. `offColor`, `onColor`, `offHandleColor` and `onHandleColor` have to be passed to the component as props so it can calculate intermediate color values. `handleDiameter`, `height` and `width` also has to be passed in as a prop since it's needed for the handle positioning calculation.

Overriding these values with `!important` can be used in some very specific cases but is generally not recommended right now.

Feel free to share specifics of what you want to do or ideas of how the component could be changed.

**Comment by Jordan1022** [comment]
I am curious how we can customize the handle itself besides onHandleColor? I would like to be able to add a border to the handle

**Comment by artemis-prime** [comment]
@Jordan1022 , this is a separate issue, no?

**Comment by Jordan1022** [comment]
@artemis-prime yes you're right
